include/preprocess/inscribed_ellipsoid_rounding.hpp

```diff
@@ -0,0 +1,159 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2020 Vissarion Fisikopoulos
+// Copyright (c) 2018-2020 Apostolos Chalkis
+
+//Contributed and/or modified by Alexandros Manochis, as part of Google Summer of Code 2020 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+
+#ifndef ELLIPSOID_ROUNDING_HPP
+#define ELLIPSOID_ROUNDING_HPP
+
+#include "max_inscribed_ellipsoid.hpp"
+#include "analytic_center_linear_ineq.h"
+#include "feasible_point.hpp"
+
+enum EllipsoidType
+{
+  MAX_ELLIPSOID = 1,
+  LOG_BARRIER = 2
+};
+
+template<int C>
+struct inscribed_ellispoid
+{
+    template<typename MT, typename Custom_MT, typename VT, typename NT>
+    inline static std::tuple<MT, VT, NT> 
+    compute(Custom_MT A, VT b, VT const& x0,
+            unsigned int const& maxiter,
+            NT const& tol, NT const& reg) 
+    {
+      std::runtime_error("No roudning method is defined");
+      return std::tuple<MT, VT, NT>();
+    }
+};
+
+template <>
+struct inscribed_ellispoid<EllipsoidType::MAX_ELLIPSOID>
```

Comment:
Not very strong opinion on this but, since we plan to implement more rounding methods I prefer to call the ellipsoid method with `inscribed_ellispoid<ellipsopid_type>::template compute<MT>` and have the specializations in structs to keep the code in the main function `inscribed_ellipsoid_rounding()` simpler and more easy to read by avoiding having a long if-else section in it.